consensus/consensus-types/src/common.rs

```diff
@@ -139,14 +139,61 @@ impl ProofWithData {
     }
 }
 
+#[derive(Deserialize, Serialize, Clone, Debug)]
+pub struct ProofWithDataV2 {
+    pub proof_with_data: ProofWithData,
+    pub txns_to_include: Option<usize>,
+}
+
+impl PartialEq for ProofWithDataV2 {
+    fn eq(&self, other: &Self) -> bool {
+        self.proof_with_data == other.proof_with_data
+            && self.txns_to_include == other.txns_to_include
+    }
+}
+
+impl Eq for ProofWithDataV2 {}
+
+impl ProofWithDataV2 {
+    pub fn new(proof_with_data: ProofWithData, txns_to_include: Option<usize>) -> Self {
+        Self {
+            proof_with_data,
+            txns_to_include,
+        }
+    }
+
+    pub fn extend(&mut self, other: ProofWithDataV2) {
```

Comment:
where is this called, what is this used for?

maybe this is not needed at all?

Reply:
This is currently called from DAG in OrderedNotifierAdapter.send_ordered_nodes(). It is implemented for Payload::InQuorumStore. So either we call it unreachable!(..) for now, either in ProofWithDataV2 or Payload,  or have some default implementation.